File: BBIS.Api/ClientApp/src/models/DonorScreening/DonorPostDonationCareDto.ts
import { IVitalSignsMonitoringDetailsDto } from "./VitalSignsMonitoringDetailsDto";
import { IPostDonationDetail,PostDonationDetailsDto } from "./PostDonationDetailsDto";

export interface IDonorPostDonationCare {
    DonorPostDonationCareId: Guid | null;
    DonorTransactionId: Guid | null;
    TypeOfReaction: string;
    SeverityOfReaction: string;
    ReactionManifestation: string;
    ActionInterventions: string;
    DoctorsNote: string;
    DischargeStatus: string;
    MonitoredBy: Guid | null;
    DoctorName: string;
    DischargeDate: Date | null;


    PostDonationListDetails: IPostDonationDetail[];
    VitalSignsMonitoringDetails: IVitalSignsMonitoringDetailsDto[];

}
export class DonorPostDonationCareDto implements IDonorPostDonationCare {
    DonorPostDonationCareId: string | null = "";
    DonorTransactionId: string | null = "";
    TypeOfReaction: string = "";
    SeverityOfReaction: string = "";
    ReactionManifestation: string = "";
    ActionInterventions: string = "";
    DoctorsNote: string = "";
    DischargeStatus: string = "";
    MonitoredBy: string | null = "";
    DoctorName: string = "";
    DischargeDate: Date | null = null

    PostDonationListDetails: IPostDonationDetail[] = [];
    VitalSignsMonitoringDetails: IVitalSignsMonitoringDetailsDto[] = [];

}

File: BBIS.Application/DTOs/DonorScreening/DonorPostDonationCareDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BBIS.Application.DTOs.DonorScreening
{
    public class DonorPostDonationCareDto
    {
        public Guid? DonorPostDonationCareId { get; set; }
        public Guid DonorTransactionId { get; set; } 
        public string TypeOfReaction { get; set; }
        public string SeverityOfReaction { get; set; }
        public string ReactionManifestation { get; set; }
        public string ActionInterventions { get; set; }
        public string DoctorsNote { get; set; }
        public string DischargeStatus { get; set; }
        public int PostDonationMonitoringId { get; set; }
        public Guid MonitoredBy { get; set; }
        public string DoctorName { get; set; }
        public DateTime DischargeDate { get; set; }

        public string DonorStatus { get; set; }

        public string DeferralStatus { get; set; }
        public string Remarks { get; set; }

        public List<PostDonationDetailsDto> PostDonationListDetails { get; set; }
        public List<VitalSignsMonitoringDto> VitalSignsMonitoringDetails { get; set; }
    }
}
